ReactAndroid/src/main/java/com/facebook/react/modules/core/DeviceEventManagerModule.java

```diff
@@ -53,6 +56,15 @@ public void emitHardwareBackPressed() {
   }
 
   /**
+   * Sends an event to the JS instance that a new intent was received.
+   */
+  public void emitNewIntent(Uri uri) {
+    getReactApplicationContext()
+        .getJSModule(RCTDeviceEventEmitter.class)
+        .emit("openURL", uri.toString());
```

Comment:
Why is this called "openURL"? If all of this logic works with generic URIs and Intents, this should be "intentReceived". Otherwise the Java method should be called `emitOpenURL` or similar.